Kim Edits-HL7-MLLP (#1324)
mulesoft/docs-connectors: hl7-mllp/2.1/modules/ROOT/edits/hl7-mllp-connector-examples.adoc

= HL7 EDI - Examples - Mule 4

> Use these examples to read and write HL7 messages over the MLLP protocol.

Reviewer: ```suggestion
Use these examples to read and write HL7 messages over Minimum Lower-Layer Protocol (MLLP).
```
Consider mentioning that there are prerequisites to using the examples and also telling what kind of example are included within.

Reviewer: Also...does one "use" examples? "follow"? Is the audience simply reading and learning from what is written here, or implementing these?

Author: Fixed. It implies using them. I think this is pretty much standard for our example docs.

> == Example: HL7 Studio

Reviewer: This title is misleading: this is not an example but rather prerequisites to following the two examples provided.

Author: Fixed.

> Set these Global properties before using the examples:

Reviewer: Why capitalize Global?

Author: fixed.

> 

Reviewer: Reformat this list to our standard for bulleted lists, per style reference.

Author: fixed.

* Delimiter: The standard HL7 delimiter value that separates

> MLLP messages. It is a hexadecimal 0x1C character also known as FS. Applications in some clinics use a different delimiter value, that can be set in this field. The standard value in the

Reviewer: ```suggestion
MLLP messages. It is a hexadecimal 0x1C character also known as FS. Applications in some clinics use a different delimiter value in this field. The standard value in the
```

Author: fixed.

HL7 MLLP connector is shown as `001C`.
+
* Host: Sets the socket host address.

> * Port: Sets the socket port.

Reviewer: ditto re. format

Author: fixed.

+
For an advanced connection, complete these fields:

> +

Reviewer: ditto format

Author: fixed.

* Reuse address: Whether to reuse the configured socket or fail when trying to bind to the socket.

> * Send TCP With No Delay: Whether or not to transmit data immediately, or collect data for greater efficiency before sending.

Reviewer: ```suggestion
* Send TCP With No Delay
+
Whether to transmit data immediately or collect data for greater efficiency before sending
```

Author: fixed

> * Linger: How long in milliseconds that the socket takes to close so any remaining data transmits correctly.

Reviewer: ```suggestion
* Linger
+
How long, in milliseconds, the socket can take to close without preventing remaining data from transmitting correctly.
```
Please verify accuracy of rewording; the original wording is awkward (how long...so data transmits...)

Author: fixed

> * Keep alive: Whether to close open socket connections that are unused for a long period and with an unavailable connection.

Reviewer: "unused...and with an unavailable connection" is awkward.
How about "unused and disconnected"?

Author: fixed

* Fail on unresolved host: Whether the socket should fail during its creation if the host set on the endpoint cannot be resolved.

> * In the TLS tab you can configure a new TLS configuration, its presence implies the use of SSLServerSocket instead of plain TCP.

Reviewer: This list item is not parallel with the previous and it is not grammatical: two sentences instead of one.

Author: rewrote

> * Set the send or receive buffer size.

Reviewer: Also not parallel with previous peer list items

Author: fixed.

== Example - Read and Write Message

> 

Reviewer: Three images without any text intro or text between them to distinguish them from one another?

Author: okay, will remove

image::hl7-mllp-demo-flow-1.png[MLLP Demo Flow in Studio 7]

image:hl7-mllp-demo-flow-2.png[MLLP Demo Flow in Studio 7]

image::hl7-mllp-demo-flow-3.png[MLLP Demo Flow in Studio 7]

> This example workflow sequentially invokes read and write HL7 messages over the MLLP protocol:

Reviewer: GLOBALLY replace "the MLLP protocol" with "MLLP"; the final P stands for protocol, so this is redundant.

Author: fixed.

> * writer-flow: Sends the given message through the socket to be received by the reader-flow.
> 
> * reader-flow: Receive a MLLP message, and log the message.

Reviewer: Reformat as proper bullet list items and remove periods; make parallel:

```suggestion
* writer-flow
+
Sends the given message through the socket to be received by the reader flow

* reader-flow
+
Receives and logs an MLLP message
```

Author: fixed.

> * The app, index, styles, app-config, rest-service, order-controller, and write-order are used to load static resources related to the user interface.

Reviewer: This list item is not parallel to the previous two
GLOBALLY make list items parallel

Author: stet

> === Example Use Case XML - read and write HL7 messages over the MLLP protocol

Reviewer: Is this supposed to be subordinate to == Example: HL7 Studio?
Why does it not use title-style capitalization throughout?
Avoid punctuation, like the "dash" here, in titles: reword for clarity

Reviewer: The second part of this existing title could probably be made a text intro to the syntax example....

Author: fixed.
